analysis/src/CompletionFrontEnd.ml

```diff
@@ -27,6 +27,38 @@ let positionToOffset text (line, character) =
     if bol + character <= String.length text then Some (bol + character)
     else None
 
+let getIdentFromExpr exp =
+  match exp.Parsetree.pexp_desc with
+  | Pexp_ident {txt} -> txt
+  | Pexp_construct ({txt}, _) -> txt
+  | Pexp_variant (label, _) -> Lident label
+  | Pexp_field (_, {txt}) -> txt
+  | _ -> Lident ""
```

Comment:
I guess this is right, because for anything else we'd like to complete from the entire context.

Reply:
Completion has a mode: field or id etc. If I remember correctly!
You might want to extract the mode.

Reply:
```ml
  (* Completion context *)
  type completionContext = Type | Value | Module | Field

```

Reply:
Btw feel free to suggest different names for the concepts.
"Context" is becoming super overloaded.

Reply:
Probably `idKind` is more fitting.

Reply:
I'll do a pass and suggest new names for a few things. I agree, context is overloaded, and there's a few similar-but-different concepts that has tripped me up a couple of times. I'll get back to you.